sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -923,7 +923,13 @@ private[client] class Shim_v2_0 extends Shim with Logging {
       tableName: String,
       throwException: Boolean): Table = {
     recordHiveCall()
-    val table = hive.getTable(dbName, tableName, throwException)
+    val table = try {
+      hive.getTable(dbName, tableName, throwException)
+    } catch {
+      // Hive may have bugs and still throw an exception even if `throwException` is false.
+      case e: HiveException if !throwException =>
```

Comment:
another idea: shall we check `tableExists` and only throw exception if table does exist?

Reply:
`tableExists` seems to depend on getTable(xx).nonEmpty?

Reply:
hmm, so `tableExists` fails as well if there is permission issues?

Reply:
Yes. Actually, if we intend to check only tableExists or not, we can revert this and restore the hive's behavior. Hive only suppresses table not found ex while throwException=true.